code/game/objects/items/devices/floorpainter.dm

```diff
@@ -0,0 +1,99 @@
+// Floor painter
+
+/obj/item/floor_painter
+	name = "floor painter"
+	icon = 'icons/obj/device.dmi'
+	icon_state = "floor_painter"
+	item_state = "electronic"
+
+	var/floor_icon
+	var/floor_state = "floor"
+	var/floor_dir = SOUTH
+
+	w_class = WEIGHT_CLASS_TINY
+	flags_1 = CONDUCT_1
+	slot_flags = ITEM_SLOT_BELT
+
+	var/static/list/allowed_states = list("arrival", "arrivalcorner", "bar", "barber", "blackcorner", "blue", "bluecorner",
+		"bluefull", "bluered", "blueyellow", "blueyellowfull", "bot", "brown", "browncorner", "browncornerold", "brownold",
+		"cafeteria", "caution", "cautioncorner", "chapel", "cmo", "dark", "delivery", "escape", "escapecorner", "floor",
+		"freezerfloor", "green", "greenblue", "greenbluefull", "greencorner", "greenfull", "greenyellow",
+		"greenyellowfull", "grimy", "loadingarea", "neutral", "neutralcorner", "neutralfull", "orange", "orangecorner",
+		"orangefull", "purple", "purplecorner", "purplefull", "rampbottom", "ramptop", "red", "redblue", "redbluefull",
+		"redcorner", "redfull", "redgreen", "redgreenfull", "redyellow", "redyellowfull", "warning", "warningcorner", "warnwhite",
+		"warnwhitecorner", "white", "whiteblue", "whitebluecorner", "whitebluefull", "whitebot", "whitecorner", "whitedelivery",
+		"whitegreen", "whitegreencorner", "whitegreenfull", "whitehall", "whitepurple", "whitepurplecorner", "whitepurplefull",
+		"whitered", "whiteredcorner", "whiteredfull", "whiteyellow", "whiteyellowcorner", "whiteyellowfull", "yellow",
+		"yellowcorner", "yellowcornersiding", "yellowsiding")
+
+/obj/item/floor_painter/afterattack(var/atom/A, var/mob/user, proximity, params)
+	if(!proximity)
+		return
+
+	var/turf/open/floor/plasteel/F = A
+	if(!istype(F))
+		to_chat(user, "<span class='warning'>\The [src] can only be used on station flooring.</span>")
+		return
+
+	F.icon_state = floor_state
+	F.icon_regular_floor = floor_state
+	F.dir = floor_dir
+
+/obj/item/floor_painter/attack_self(var/mob/user)
+	if(!user)
+		return 0
+	user.set_machine(src)
+	interact(user)
+	return 1
+
+/obj/item/floor_painter/interact(mob/user as mob)
+	if(!floor_icon)
+		floor_icon = icon('icons/turf/floors.dmi', floor_state, floor_dir)
+	user << browse_rsc(floor_icon, "floor.png")
+	var/dat = {"
+		<center>
+		<a href="?src=[UID()];cycleleft=1">&lt;-</a>

```

Comment:
Replace `<a href="?src=[UID()];cycleleft=1">&lt;-</a>` with `<a href="?src=[RED(src)];cycleleft=1">&lt;-</a>`, and the same for the ones below